main.c

```diff
@@ -14,15 +14,59 @@ how to use the page table and disk interfaces.
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
+#include <stdbool.h>
 
-void page_fault_handler( struct page_table *pt, int page )
-{
-	printf("page fault on page #%d\n",page);
-	exit(1);
+struct disk* disk;
+
+void evictPage(struct page_table *pt, int page);
+
+void loadFrameIntoPage(struct page_table *pt, int page, int frame);
+
+void page_fault_handler(struct page_table *pt, int page ) {
+    static int curr_frame = 0;
+    int frame = -1;
+    for(int f=0;f<pt->nframes;f++){
+
+	// Find a frame that is not already allocated to a page.
+	bool success = true;
+	for (int curr_page=0; curr_page < pt->npages; curr_page++){
+	    if(pt->page_mapping[curr_page] == f && pt->page_bits[curr_page] != 0){
+		success = false;
+		break;
+	    }
+	} if(success){
+	    frame = f;
+	    break;
+	}
+    }
+    if(frame == -1){
+        // Pick a random page that owns a frame.
+        int curr_page;
+       
+	curr_page = curr_frame % pt->npages;
+
```

Comment:
This will always evict the oldest page

Reply:
I just realized that what I implemented was a LIFO queue. This is a good solution.